packages/components/src/components/stateful-providers/RuleOptionsProvider/RuleOptionsProvider.tsx

```diff
@@ -153,18 +153,46 @@ function RuleOptionsProvider(props: IProps): JSX.Element {
             }
 
             // optionLabelFilters.catalog = `/localized_catalogs/${localizedCatalogId}`
+            let currentPage = 0
+            const limit = 50
+
             const optionsResponse = await fetchApi<
               IHydraResponse<ISourceFieldOption>
             >(
               sourceFieldOptionResource,
-              getListApiParameters(false, undefined, optionLabelFilters)
+              getListApiParameters(currentPage, limit, optionLabelFilters)
             )
 
             if (!isError(optionsResponse)) {
-              return getOptionsFromOptionResource(
+              let totalItems = optionsResponse['hydra:totalItems']
+              let options = getOptionsFromOptionResource(
                 optionsResponse,
                 localizedCatalogId
               )
+              /* eslint-disable no-await-in-loop */
+              while (totalItems > (currentPage + 1) * limit) {
+                currentPage++
+                const nextOptionsResponse = await fetchApi<
+                  IHydraResponse<ISourceFieldOption>
+                >(
+                  sourceFieldOptionResource,
+                  getListApiParameters(currentPage, limit, optionLabelFilters)
+                )
+                if (!isError(nextOptionsResponse)) {
+                  totalItems = nextOptionsResponse['hydra:totalItems']
+                  options = [
+                    ...options,
+                    ...getOptionsFromOptionResource(
+                      nextOptionsResponse,
+                      localizedCatalogId
+                    ),
+                  ]
+                } else {
+                  throw new Error('error')
+                }
+              }
+              /* eslint-enable no-await-in-loop */
```

Comment:
you can refactor your code to call once `await fetchApi<IHydraResponse<ISourceFieldOption>>(...` by using a "do while"
```
do {
//my calls to get options
} while (condition)
```

for eslint-enable no-await-in-loop check this article https://medium.com/@chrisjr06/why-and-how-to-avoid-await-in-a-for-loop-32b86722171